src/Castle.Transactions/Internal/TransactionLogic.cs

```diff
@@ -47,6 +47,7 @@ public static Func<IDisposable> ForkScopeFactory(this ITransactionManager manage
 		{
 			return () =>
 			{
+				manager.Activities.CreateNewActivity();

```

Comment:
If `CreateNewActivity` returned the activity it created you wouldn't need to call `GetCurrentActivity` twice here.